docs: add contributor documentation

MartinCupela opened the pull request:
### 🎯 Goal

Provide an onboarding guide for developers that would like to start contributing to the `stream-chat-react` repository.

As the repository developers come and go, the domain knowledge tends to disappear. This documentation should provide single source of truth on how to maintain a certain level of the repository's quality standard. It should represent a shared contract removing any doubts about how the things should be done correctly.

### 🛠 Implementation details

The main file is `CONTRIBUTING.md`. It then refers to more narrow-scoped files in `developers` directory.

fixes #1416

arnautov-anton commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,77 @@
+Keep in mind that the chat application may need to work with thousands of messages.
+
+#### 5. Follow commit formatting rules
+We follow [Angular's Commit Message Format rules](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-format) with [possible deviations](./developers/COMMIT.md). The same rules are used by our release automation tool. Therefore, every commit message should strictly comply with these rules.
```
```suggestion
We follow [Angular's Commit Message Format rules](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-format) with [possible deviations](./developers/COMMIT.md). The same rules are used by our release automation tool. Therefore, every commit message should strictly comply with these rules. If you're still unsure, you can use [commitizen](https://www.npmjs.com/package/commitizen) as your guide.
```

petyosi replied:
We need to integrate commitlint. React Native has it.

MartinCupela replied:
Integrating commitlint

petyosi commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,77 @@
+
+
+## <a name="asking-questions"></a>Are you looking for answers?
+There are many ways you can get your questions answered. It can be hard to decide, where to begin if you are just starting out. We suggest you take a look at the resources in the following order
```
```suggestion
There are many ways you can get your questions answered. It can be hard to decide, where to begin if you are just starting out. We suggest you take a look at the resources in the following order:
```

MartinCupela replied:
Fixing typo

petyosi commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,77 @@
+### Take a look at our tutorials
+
+### Get help from our Customer Success team
+If what you are looking for is technical support embedding the Stream's chat in your application, we would suggest emailing our Customer Success and Engineer support team at support@getstream.io with your application key or ID. The Issue section of this GitHub repo is now reserved only for bug reports, feature improvements, and suggestions.
```
```suggestion
If what you are looking for is technical support embedding Stream in your application, we suggest emailing our Customer Success team at support@getstream.io with your application key and the SDK versions you're using. The Issue section of this GitHub repo is now reserved only for bug reports, feature improvements, and suggestions.
```

MartinCupela replied:
Adding the suggestion to update the section Get help from our Customer Success team

szuperaz commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,45 @@
+
+**Additional context**
+
+Add any other context about the problem here.
```
I think stream-chat-react and stream-chat-js version should be useful as well, it should be in the `x-stream-client` header of requests (however it seems to me that right now the version is '0.0.0-development' for React, I'll post a Slack message about this)

MartinCupela replied:
Added a section Package version to the bug template. Thank you

MartinCupela commented:
@petyosi, @arnautov-anton @szuperaz  I have added `commitlint` to check the commit message format before on each commit. Originally I planned to follow the [commitlint documentation](https://commitlint.js.org/#/guides-local-setup) and upgrade husky as well. This turned out to be a bad idea. 

**Lesson learned:**
`stream-chat-react` has `husky v4.3.8` installed, but `stream-chat-react-native` has husky v6. The v6 contains major braking changes and introduces a need to add `postinstall` script (`postinstall: husky install`) into the `package.json` in order the husky is set up on `git clone ... && yarn install`. What `husky install` under the hood does, it changes the default `git config core.hooksPath` (path to the hooks directory = `.git/hooks`) to `.husky`. As `postinstall` is run when `stream-chat-react` is downloaded as a dependency by our users, it would require another dependency (`pinst`) to disable `postinstall` to be run if downloaded as dependency (= addition of scripts `prepack: pinst --disable`, `postpack: pinst --enable`).

If `postinstall` is not included among the scripts in `package.json` then the `husky install` is not automatically run and then there are no hooks set up in recently cloned repository.

I do not like the idea of installing dependencies just because the authors of our tools took bad design decisions.


And so I kept `husky v4.3.8` as it sets up .git/hooks directory (populates it with necessary scripts) on `yarn install` run by `stream-chat-react` developer after cloning the repo.

I have added missing commit types supported by `commitlint`'s default config to `developers/COMMIT.md`.

petyosi commented:
:tada: This PR is included in version 8.1.1 :tada:

The release is available on:
- [GitHub release](https://github.com/GetStream/stream-chat-react/releases/tag/v8.1.1)
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/stream-chat-react/v/8.1.1)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
